# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546405544 28800
#      Tue Jan 01 21:05:44 2019 -0800
# Branch stable
# Node ID d43ba2c21d1da9193d795fdd5f64f6f2998d9668
# Parent  8020c9ba82fcc710c5faf964bc55ef830754c0ff
ordschur.cc: Fix static analyzer detected issues (bug #55347).

* ordschur.cc (Fordschur): Call args(2).xoctave_idx_type_vector_value()
rather than incorrect version without prepended 'x'.

diff --git a/libinterp/corefcn/ordschur.cc b/libinterp/corefcn/ordschur.cc
--- a/libinterp/corefcn/ordschur.cc
+++ b/libinterp/corefcn/ordschur.cc
@@ -72,17 +72,17 @@ is in the upper left corner, by doing:
 @end example
 
 @seealso{schur, ordeig}
 @end deftypefn */)
 {
   if (args.length () != 3)
     print_usage ();
 
-  const Array<octave_idx_type> sel_arg = args(2).octave_idx_type_vector_value ("ordschur: SELECT must be an array of integers");
+  const Array<octave_idx_type> sel_arg = args(2).xoctave_idx_type_vector_value ("ordschur: SELECT must be an array of integers");
 
   const octave_idx_type sel_n = sel_arg.numel ();
 
   const dim_vector dimU = args(0).dims ();
   const dim_vector dimS = args(1).dims ();
 
   if (sel_n != dimU(0))
     error ("ordschur: SELECT must have same length as the sides of U and S");
